week2/index.js

```diff
@@ -0,0 +1,119 @@
+Added Abdullah
+
+"use strict";
+
+const fs = require('fs');
+const TODOS_FILENAME = process.env["TODOS_FILENAME"] || "todos.json";
+const helpFile = "help.txt";
+
+let commandArgs = process.argv.slice(2);
+let myTodoList = [];
+let num = 1;
+
+
+fs.readFile(TODOS_FILENAME, "utf8", (err, data) => {
+
+    if (err) {
+        //console.log(err);
+        if (err.code === "ENDENT") {
+
+            //No such File or directory
+            console.log("i'm sorry, friend, but that file doesn't exist.");
+        }
+        else if (err.code === "EACCES") {
+            console.log("That file exists but you don't have permission.");
+
+        }
+        else {
+
+            console.log(`Uh oh, an unknown error occured ${err.message}`)
+        }
+        process.exit();
+
+    }
+
+    // console.log(`Successfully read the file: ${TODOS_FILENAME}`)
+
+    myTodoList = JSON.parse(data);
+
+    if (commandArgs[0] === "add") {
+
+        addItem();
```

Comment:
A better pattern would be to create functions that take a parameter so that you call them with `addItem(itemAsString)`, for example. That way the function doesn't rely on global variables.

Reply:
Aha, thanks good to know that.